### DISCUSSION TITLE
The SQL model cannot perform insertion into another table using queried data

### DISCUSSION THREAD

**Discussion by writebai2** [fastapi/sqlmodel#1232] [Questions] [answered]
### Issue Content

The SQL model cannot insert the queried data into the corresponding table of another database

```python

engine = create_engine(str(settings.MYSQL_DATABASE_URL))
rds_engine = create_engine(str(settings.RDS_DATABASE_URI), echo=True)

def get_online_products():
    statement = (
        select(
            Tbl_Product
        )
        .limit(1)
    )

    with Session(engine) as session:
        products = session.exec(statement).all()
    return products

def insert_main():
    product_datas = get_online_products()
    with Session(rds_engine) as session:
        session.add_all(product_datas)
        session.commit()

if __name__ == "__main__":
    insert_main()
```

After execution：
PS D:\python\db_visualization> D:\python\myvenv\Scripts\python.exe "d:\python\db_visualization\back_main.py"

There is no response in the table after execution

**Comment by tholo** [comment]
I forget if this is the case with plain SQLAlchemy -- if so, you may have to `expunge()` them (or all with `expunge_all()`) from the current session before adding it to a different one...?

**Comment by writebai2** [comment, marked as answer]
> I forget if this is the case with plain SQLAlchemy -- if so, you may have to expunge() them (or all with expunge_all()) from the current session before adding it to a different one...?

I solved this problem by using mapping to insert data instead. Thank you
